### DISCUSSION TITLE
Update the firmware using VAPIX API

### DISCUSSION THREAD

**Discussion by vivekatoffice** [AxisCommunications/discussions#123] [VAPIX] [answered]
As per the VAPIX documentation [Upgrade firmware](https://www.axis.com/vapix-library/subjects/t10175981/section/t10118139/display?section=t10118139-t10118132) we understand the firmware file can be uploaded to Axis device:

```curl
//<servername>/axis-cgi/firmwaremanagement.cgi

POST /axis-cgi/firmwaremanagement.cgi HTTP/1.1
Content-Type: multipart/form-data; boundary=<boundary>
Content-Length: <content length>

--<boundary>
Content-Type: application/json

{
  "apiVersion": "1.0",
  "context": "abc",
  "method": "upgrade"
}

--<boundary>
Content-Type: application/octet-stream

<firmware file content>

--<boundary>--

```

Please provide samples or cURL command for this if you have successfully implemented. 

Please find my command(**Not working**) below:  

```curl
 curl -u root:pass --digest --noproxy "*" --request POST 'http://195.60.68.14/axis-cgi/firmwaremanagement.cgi' 
--header 'Content-Type: multipart/form-data; boundary=--axis' 
--form '--axis' 
--form 'Content-Type="application/json"' 
--form '{ "apiVersion"=" \"1.0\",  \"context\": \"abc\", \"method\": \"upgrade\"}"' 
--form '--axis' 
--form 'Content-Type="application/octet-stream"' 
--form 'filename2="1.bin"' 
--form '--axis' 
--form '1.bin=@"1.bin"'

```

**Comment by mattias-kindborg-at-work** [comment, marked as answer]
Hi @axisvivek! I think the following curl command will work.

```sh
curl --anyauth -u "root:<device password>" \
  -F 'body={"apiVersion":"1.0","context":"my-context","method":"upgrade"}' \
  -F packfil=@Q1656_11_0_108.bin "<device ip>/axis-cgi/firmwaremanagement.cgi"
```

where `<device password>` is the device password, `Q1656_11_0_108.bin` is the firmware file in current directory, and `<device ip>` is the IP adress of the device.

By the way, `\` are indicators on Linux and macOS that the command is continuing on the next line, on Windows you can simply omit them.

**Reply by vivekatoffice** [reply]
Thank you so much for the help.

**Reply by vivekatoffice** [reply]
![image](https://user-images.githubusercontent.com/52032125/206443659-70ba94b1-3fdc-4c7e-8354-a28adba8cad4.png)

Successful response 👍

**Comment by wiener-dog** [comment]
Worked great for me.  I  just added the -content-type: multipart/form-data option.  Thank  you @mattias-kindborg-at-work and @vivekatoffice, you made my life much easier.  I can now fully deploy cameras from my script.  Minus the initial root account setup.  That still doesn't work.

**Reply by mattias-kindborg-at-work** [reply]
Do you have a discussion open for the initial root account setup?

**Reply by wiener-dog** [reply]
Yes I do.  Its here.  https://github.com/orgs/AxisCommunications/discussions/622

I have some spare cycles today, so might spend some time on it.
